File: mobile/packages/footprint_flutter/lib/src/onboarding-components/models/footprint_error.dart
enum ErrorKind {
  initializationError,
  authError,
  invalidAuthTokenError,
  userError,
  decryptionError,
  vaultingError,
  onboardingError,
  inlineProcessNotSupported,
  inlineOtpNotSupported,
  notAllowed,
  webviewError,
  uiError,
}

class FootprintError implements Exception {
  final ErrorKind kind;
  final String message;
  final String? supportId;
  final Map<String, String>? context;

  FootprintError({
    required this.kind,
    required this.message,
    this.supportId,
    this.context,
  });

  @override
  String toString() {
    String contextStr =
        context != null && context!.isNotEmpty ? ', context: $context' : '';
    return 'FootprintError(kind: $kind, message: $message$contextStr, supportId: $supportId)';
  }
}

File: Sources/Footprint/FootprintError.swift
public struct FootprintError {
    public enum ErrorKind {
        case initializationError
        case authError
        case userError
        case decryptionError
        case vaultingError(context: [String: String]?)
        case onboardingError
        case inlineProcessNotSupported
        case inlineOtpNotSupported
        case notAllowed
        case webviewError
        case uiError
    }
    
    public let kind: ErrorKind
    public let message: String
}
